rental-car-manager/editorextensions/rental-car-manager/src/rules/color.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ConditionType,
+  ShapeDataConditions,
+  ConditionCombination,
+  RuleFormattingType,
+  DataGraphicIconSets,
+  HorizontalBadgePos,
+  VerticalBadgePos,
+  BadgeLayerPos,
+  BadgeResponsiveness,
+} from "lucid-extension-sdk";
+import { RuleDefinition } from "lucid-extension-sdk/document/ruledefinition";
+
+export const carColorRuleDefinition: RuleDefinition = {
+  name: "Car color icon",
+  effects: [
+    {
+      conditions: [
+        {
+          type: ConditionType.ShapeData,
+          condition: ShapeDataConditions.Equal,
+          value: "Red",
+          field: "Color Name",
+        },
+      ],
+      combination: ConditionCombination.AND,
+      formatType: RuleFormattingType.DATA_GRAPHICS,
+      dataGraphic: {
+        set: DataGraphicIconSets.STOPLIGHTS,
+        index: 0,
+        color: "#e81313",
+        position: {
+          horizontalPos: HorizontalBadgePos.RIGHT,
+          verticalPos: VerticalBadgePos.TOP,
+          layer: BadgeLayerPos.INSIDE,
+          responsive: BadgeResponsiveness.STACK,
+        },
+      },
+      tooltip: '=@"Color Name"',
+    },
+    {
+      conditions: [
+        {
+          type: ConditionType.ShapeData,
+          condition: ShapeDataConditions.Equal,
+          value: "White",
+          field: "Color Name",
+        },
+      ],
+      combination: ConditionCombination.AND,
+      formatType: RuleFormattingType.DATA_GRAPHICS,
+      dataGraphic: {
+        set: DataGraphicIconSets.STOPLIGHTS,
+        index: 0,
+        color: "#ffffff",
+        position: {
+          horizontalPos: HorizontalBadgePos.RIGHT,
+          verticalPos: VerticalBadgePos.TOP,
+          layer: BadgeLayerPos.INSIDE,
+          responsive: BadgeResponsiveness.STACK,
+        },
+      },
+      tooltip: '=@"Color Name"',
+    },
+    {
+      conditions: [
+        {
+          type: ConditionType.ShapeData,
+          condition: ShapeDataConditions.Equal,
+          value: "Gray",
+          field: "Color Name",
+        },
+      ],
+      combination: ConditionCombination.AND,
+      formatType: RuleFormattingType.DATA_GRAPHICS,
+      dataGraphic: {
+        set: DataGraphicIconSets.STOPLIGHTS,
+        index: 0,
+        color: "#979ea8",
+        position: {
+          horizontalPos: HorizontalBadgePos.RIGHT,
+          verticalPos: VerticalBadgePos.TOP,
+          layer: BadgeLayerPos.INSIDE,
+          responsive: BadgeResponsiveness.STACK,
+        },
+      },
+      tooltip: '=@"Color Name"',
+    },
```

Comment:
nit: Might be able to deduplicate a lot of the attributes that are the same across all three colors. Would help to explain what's going on here